mlrun/model_monitoring/db/tsdb/timescaledb/queries/timescaledb_results_queries.py

```diff
@@ -0,0 +1,583 @@
+
+        query = table_schema._get_records_query(
+            start=start,
+            end=end,
+            columns_to_filter=columns,
+            filter_query=filter_query,
+            group_by=group_by_columns,
+            order_by=order_by_clause,
+        )
+
+        result = self._connection.run(query=query)
+
+        if not result or not result.data:
+            return {}
+
+        return {(row[0], row[1]): row[2] for row in result.data}
+
+    def get_drift_data(
+        self,
+        start: datetime,
+        end: datetime,
+        interval: Optional[str] = None,
+    ) -> mm_schemas.ModelEndpointDriftValues:
+        """
+        Get drift data aggregated by time intervals, showing the count of suspected and detected drift events.
+
+        This method queries the app_results table for drift-related statuses (potential_detection=1, detected=2)
+        and aggregates them by time intervals, counting the maximum drift status per endpoint per interval.
+
+        :param start: Start time for the query
+        :param end: End time for the query
+        :param interval: Optional time interval for aggregation (e.g., "1 hour", "30 minutes").
+                        If not provided, will be automatically determined based on query duration.
+        :return: ModelEndpointDriftValues containing time-binned drift counts
+        """
+        # Align start/end times and determine interval
+        start, end = self._pre_aggregate_handler.get_start_end(start, end)
+
+        if interval is None:
+            # Automatically determine appropriate interval based on time range
+            interval = TimescaleDBQueryBuilder.determine_optimal_interval(start, end)
+
+        # Use pre-aggregate handler to align times to interval boundaries
+        start, end = self._pre_aggregate_handler.align_time_range(start, end, interval)
+
+        # Build status filter for drift-related statuses only
+        suspected_status = mm_schemas.ResultStatusApp.potential_detection.value  # 1
+        detected_status = mm_schemas.ResultStatusApp.detected.value  # 2
+
+        app_results_table = self.tables[mm_schemas.TimescaleDBTables.APP_RESULTS]
+
+        # Use TimescaleDB's time_bucket function for interval aggregation
+        query = f"""
+        WITH drift_intervals AS (
+            SELECT
+                time_bucket('{interval}', {mm_schemas.WriterEvent.END_INFER_TIME}) AS bucket_start,
+                {mm_schemas.WriterEvent.ENDPOINT_ID},
+                MAX({mm_schemas.ResultData.RESULT_STATUS}) AS max_status
+            FROM {app_results_table.full_name()}
+            WHERE {mm_schemas.ResultData.RESULT_STATUS} IN (%s, %s)
+            AND {mm_schemas.WriterEvent.END_INFER_TIME} >= %s
+            AND {mm_schemas.WriterEvent.END_INFER_TIME} <= %s
+            GROUP BY bucket_start, {mm_schemas.WriterEvent.ENDPOINT_ID}
+        )
+        SELECT
+            bucket_start,
+            max_status,
+            COUNT(*) as status_count
```

Comment:
For consistency, applies elsewhere as well:
```suggestion
            COUNT(*) AS status_count
```